hbase-server/src/main/java/org/apache/hadoop/hbase/master/normalizer/SimpleRegionNormalizer.java

```diff
@@ -468,12 +467,13 @@ private static boolean isOldEnoughForMerge(
    */
   private boolean isLargeEnoughForMerge(
     final NormalizerConfiguration normalizerConfiguration,
+    final NormalizeContext ctx,
     final RegionInfo regionInfo
   ) {
-    return getRegionSizeMB(regionInfo) >= normalizerConfiguration.getMergeMinRegionSizeMb();
+    return getRegionSizeMB(regionInfo) >= normalizerConfiguration.getMergeMinRegionSizeMb(ctx);
   }
 
-  private static boolean logTraceReason(final BooleanSupplier predicate, final String fmtWhenTrue,
+  private boolean logTraceReason(final BooleanSupplier predicate, final String fmtWhenTrue,
```

Comment:
why make this an instance method?

Reply:
I find skipForSplit, logTraceReason, isOldEnoughForMerge is static, but it does not need static. I am not sure what is the purpose of setting those to static.

Reply:
As a general practice, when a method can be static, I prefer to make it so. If a method has no need for `this` or private state, why expose it? I find static methods are easier to reason about because you know there's no hidden effects of object member variables manipulated elsewhere. I suspect they're simpler for the JIT to optimize for the same reasons, but I don't have evidence of this.

Reply:
Thanks for the wonderful answers, which benefited me a lot.